setup.py

```diff
@@ -63,9 +63,11 @@
     "bignumeric_type": pyarrow_dep,
     "tqdm": ["tqdm >= 4.7.4, <5.0.0dev"],
     "opentelemetry": [
-        "opentelemetry-api >= 0.11b0",
-        "opentelemetry-sdk >= 0.11b0",
-        "opentelemetry-instrumentation >= 0.11b0",
+        # The Opentelemetry API was changed in v0.15b, we need to bound the version.
+        # https://github.com/open-telemetry/opentelemetry-python/commit/f3cdfa2cdb9cb1c442189e2ead3788f45d92352d#diff-ebe3b7e2373ce704ff49dcab7a82d2270aa35c624337325cd0233e60e7a95a39R35
+        "opentelemetry-api >= 0.11b0, <0.15b0",
+        "opentelemetry-sdk >= 0.11b0, <0.15b0",
+        "opentelemetry-instrumentation >= 0.11b0, <0.15b0",
```

Comment:
I discovered this when investigating the alternating `mypy` error and unit test error under Python 3.6 (fixing one produced the other). It turned out that the test/check failure depends on which `opentelemetry-api` version is installed, because the latter's surface has changed since `0.11b` that we pin as a minimum.

We should probably bump the opentelemetry version in a separate PR, though. We would like to use the more up-to-date API, should preferably also be a version that is type-checked (at least [1.1.0](https://github.com/open-telemetry/opentelemetry-python/blob/main/CHANGELOG.md#110---2021-04-20)).

Reply:
+1 on bumping opentelemetry in a separate PR.

Note: we might want to consider #1038 to be a bug, since I think it means we can't have Spanner and BigQuery clients installed in the same environment.